src/libostree/ostree-bootloader-grub2.c

```diff
@@ -315,7 +316,45 @@ _ostree_bootloader_grub2_write_config (OstreeBootloader      *bootloader,
         use_system_grub2_mkconfig = FALSE;
     }
   else
-    grub_exec = use_system_grub2_mkconfig ? GRUB2_MKCONFIG_PATH : "/usr/lib/ostree/ostree-grub-generator";
+    {
+      if (use_system_grub2_mkconfig)
+        {
+          grub_exec = GRUB2_MKCONFIG_PATH;
+        }
+      else
+        {
+          g_autoptr(GPtrArray) boot_loader_configs = NULL;
+          OstreeBootconfigParser *config;
+          g_autofree char *ostree_arg = NULL;
+          g_autofree char *sysroot_abspath = g_file_get_path (self->sysroot->path);
+          struct stat stbuf;
+
+          if (!_ostree_sysroot_read_boot_loader_configs (self->sysroot, bootversion, &boot_loader_configs,
+                                                         cancellable, error))
+            goto out;
+
+          config = boot_loader_configs->pdata[0];
+          ostree_arg = get_ostree_kernel_arg_from_config (config);
+          if (!ostree_arg)
+            {
+              g_set_error_literal (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                                   "No ostree= kernel argument found in boot loader configuration file");
+              goto out;
+            }
+
+          snprintf (generator_path, sizeof(generator_path), "%s/%s/usr/lib/ostree/ostree-grub-generator",
```

Comment:
> I have pending work on recompiling SELinux policy, and was going to look at doing that via a container (bwrap/systemd-nspawn) in the new root.

I have a script that I use for modifying ostree images by running a command in them: https://gist.github.com/wmanley/243cbcbe7c3cb938acc4cf5d689fde22 .  I use it with `qemu-user-static` so I can run commands on my ARMv7 images on my AMD64 PC.  It uses `runc`.  It may be useful to you.